polyfill/lib/absolute.mjs

```diff
@@ -16,39 +16,33 @@ import {
   NANOSECONDS
 } from './slots.mjs';
 import { absolute as STRING } from './regex.mjs';
+import bigInt from 'big-integer';
 
 export class Absolute {
   constructor(epochNanoseconds) {
-    if ('bigint' !== typeof epochNanoseconds) throw RangeError('bigint required');
-    const epochMilliseconds = Number(epochNanoseconds / BigInt(1e6));
-    const restNanoseconds = Number(epochNanoseconds % BigInt(1e6));
+    if (('bigint' !== typeof epochNanoseconds) && !bigInt.isInstance(epochNanoseconds)) throw RangeError('bigint required');
     CreateSlots(this);
-    SetSlot(this, EPOCHNANOSECONDS, { ms: epochMilliseconds, ns: restNanoseconds });
+    SetSlot(this, EPOCHNANOSECONDS, bigInt(epochNanoseconds));
   }
 
   getEpochSeconds() {
     if (!ES.IsAbsolute(this)) throw new TypeError('invalid receiver');
     const value = GetSlot(this, EPOCHNANOSECONDS);
-    const epochSeconds = Math[value.ms < 0 ? 'ceil' : 'floor'](value.ms / 1000);
-    return epochSeconds;
+    return value.divide(1e9);
   }
   getEpochMilliseconds() {
     if (!ES.IsAbsolute(this)) throw new TypeError('invalid receiver');
-    const value = GetSlot(this, EPOCHNANOSECONDS);
-    const epochMilliSeconds = value.ms;
-    return epochMilliSeconds;
+    const value = bigInt(GetSlot(this, EPOCHNANOSECONDS));
+    return value.divide(1e6);
   }
   getEpochMicroseconds() {
     if (!ES.IsAbsolute(this)) throw new TypeError('invalid receiver');
-    const value = GetSlot(this, EPOCHNANOSECONDS);
-    const epochNanoseconds = BigInt(value.ms) * BigInt(1e6) + (BigInt(value.ns) % BigInt(1e6));
-    return epochNanoseconds / BigInt(1e3);
+    const value = bigInt(GetSlot(this, EPOCHNANOSECONDS));
+    return value.divide(1e3).value;
```

Comment:
```suggestion
    return value.divide(1e3);
```

Reply:
This would be wrong, because it has to return an actual BigInt, not the wrapped version the big-integer polyfill produces.